File: src/main/java/com/example/gql_demo/BookFilter.java
package com.example.gql_demo;

import java.util.List;

// inspired by https://github.com/intuit/graphql-filter-java
public record BookFilter(
    StringExpression name,
    List<BookFilter> and,
    List<BookFilter> or,
    List<BookFilter> not
) {
}

File: src/main/java/com/example/gql_demo/BookFilter.kt
package com.example.gql_demo

// inspired by https://github.com/intuit/graphql-filter-java
data class BookFilter(
    val name: StringExpression?,
    val and: List<BookFilter>?,
    val or: List<BookFilter>?,
    val not: List<BookFilter>?
)

